src/main/java/com/github/checkstyle/regression/configuration/ConfigGenerator.java

```diff
@@ -0,0 +1,211 @@
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.configuration;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import com.github.checkstyle.regression.data.ModuleInfo;
+
+/**
+ * Generates the config XML and output it to a specific file.
+ * @author LuoLiangchen
+ */
+public final class ConfigGenerator {
+    /** The parent name "Checker". */
+    private static final String PARENT_CHECKER = "Checker";
+
+    /** The parent name "TreeWalker". */
+    private static final String PARENT_TREE_WALKER = "TreeWalker";
+
+    /** The name of a module element. */
+    private static final String ELEMENT_MODULE = "module";
+
+    /** The name of a property element. */
+    private static final String ELEMENT_PROPERTY = "property";
+
+    /** The attribute name "name". */
+    private static final String ATTR_NAME = "name";
+
+    /** The attribute name "value". */
+    private static final String ATTR_VALUE = "value";
+
+    /** The "doctype-public" value of the config. */
+    private static final String DOCTYPE_PUBLIC =
+            "-//Puppy Crawl//DTD Check Configuration 1.3//EN";
+
+    /** The "doctype-system" value of the config. */
+    private static final String DOCTYPE_SYSTEM =
+            "http://www.puppycrawl.com/dtds/configuration_1_3.dtd";
+
+    /** The config template. */
+    private static final String BASE_CONFIG = ""
+            + "<?xml version=\"1.0\" standalone=\"yes\"?>"
+            + "<module name=\"Checker\">\n"
+            + "  <property name=\"charset\" value=\"UTF-8\"/>\n"
+            + "  <property name=\"severity\" value=\"warning\"/>\n"
+            + "  <property name=\"haltOnException\" value=\"false\"/>\n"
+            + "  <module name=\"BeforeExecutionExclusionFileFilter\">\n"
+            + "    <property name=\"fileNamePattern\" value=\"module\\-info\\.java$\" />\n"
+            + "  </module>\n"
+            + "  <!-- Contents below are generated by regression-tool. -->\n"
+            + "  <module name=\"TreeWalker\"/></module>";
```

Comment:
> Someday we may add other TreeWalkers

I don't know whats the meaning of this, maybe smth like `MutiThreadTreeWalker` in the future?

I think by now, it is OK to let the tool generate `TreeWalker` dynamically. But there would be some other questions:
If we assume that the module would have whatever parent(not just `Checker` and `TreeWalker`), so how to decide where to put the parent module? Should we gain the info from the module extract info map? If yes, then configuration component would depend on module component, otherwise we would have to pass the whole info map to configuration component, but I don't think it is a good idea.